addons/mass_mailing/static/src/js/mass_mailing_html_field.js

```diff
@@ -255,6 +255,7 @@ export class MassMailingHtmlField extends HtmlField {
         const $themeSelectorNew = $(qweb.render("mass_mailing.theme_selector_new", {
             themes: themesParams,
             templates: templatesParams,
+            validTemplates: templatesParams.filter(template => template.modelId === this.props.record.data.mailing_model_id[0]),
```

Comment:
Hello @thiryr 

Do not remember seeing this the first time.

What does "validTemplates" imply exactly?
It seems like the xml loops on the "templates" key anyway, so I'm not sure I understand the distinction.

Reply:
I didn't take into account templates existing but not being displayed (because not the same 'model')

The 'invalid' templates are hidden when ``_hideIrrelevantTemplates`` is called, not sure why we're not filtering templates directly already (maybe for dynamic updates without using states, since it's rendered manually?).

I need this otherwise if there exists ANY template, the div is rendered with a space and we have a 1-character-high gap.


I could probably add the logic for this inside ``_hideIrrelevantTemplates`` to keep things in one place though.

Reply:
I see there is indeed some specific logic in `_hideIrrelevantTemplates`.

What we could do is keep the code as you had before, but add logic into `_hideIrrelevantTemplates` to hide this wrapper div when it shows the "o_mailing_template_message" (and invert the logic in the else case).

This would require to add a specific class to target that div, something like "o_mailing_template_preview_wrapper".

That way we keep it simple and integrate into the existing code.

Does that make sense?

PS: Sorry for the delayed answer, it's crazy times with the v16 """freeze""" ^^

Reply:
^^ no problem, this is very far from my oldest PR anyway.

I moved the logic inside ``_hideIrrelevantTemplates`` as discussed.